practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/configuration/PractitionerConfigurationProvider.java

```diff
@@ -1,38 +1,58 @@
 package uk.ac.ox.ndph.mts.practitioner_service.configuration;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.core.io.Resource;
+import org.springframework.http.MediaType;
 import org.springframework.stereotype.Component;
-
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientException;
+import uk.ac.ox.ndph.mts.practitioner_service.exception.RestException;
 import uk.ac.ox.ndph.mts.practitioner_service.model.PractitionerConfiguration;
-import uk.ac.ox.ndph.mts.practitioner_service.exception.InitialisationError;
+import java.io.File;
 
 /**
  * Provide Practitioner Configuration 
  */
 @Component
 public class PractitionerConfigurationProvider {
 
-    @Value("classpath:practitioner-configuration.json")
-    Resource configurationFile;
+    @Value("${spring.cloud.config.uri}")
+    private String configURI;
+
+    @Value("${spring.application.name}")
+    private String applicationName;
+
+    @Value("${spring.cloud.config.profile}")
+    private String profile;
+
+    @Value("${spring.cloud.config.label}")
+    private String label;
 
-    private PractitionerConfiguration configuration;
 
     /**
      * Get Practitioner Configuration
      * @return PractitionerConfiguration
      */
     public PractitionerConfiguration getConfiguration() {
-        if (configuration == null) {
-            try {
-                String jsonString = new String(configurationFile.getInputStream().readAllBytes());
-                configuration = new ObjectMapper().readValue(jsonString, PractitionerConfiguration.class);
-            } catch (Exception e) {
-                throw new InitialisationError(Configurations.ERROR.message(), e);
-            }
+
+        PractitionerConfiguration practitionerConfiguration;
+
+        try {
```

Comment:
how would that work locally?
will it just use the git as well? or can we have also support for local file for these cases?

Reply:
can you add this comment:
// Asking the configuration server for the json file

Reply:
why do we want it to work 'locally' - i,e with resources - I thought the point of having the config server was to decentralize config files?

Reply:
you are right its not the place I meant to comment in.
what I meant is that the config server will , in local mode, wont go to git, instead it will use a file...

Reply:
why (in what scenarios) would we use 'local' resources when using the config server?